### PR TITLE
Add Votes feature

### PR DESCRIPTION
**Pull request by HuitingZhang**
Implemented function in front-end, the swipe effect have not done yet.

### CODE REVIEW

File: app/assets/javascripts/components/post.js.jsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+var Score = require('./score');
 
 var Post = React.createClass({
   propTypes: {
-    post: React.PropTypes.object,
+    post: React.PropTypes.object
   },
 
   render: function() {
+      console.log(this.props.post);
```
Comment (bradencanderson) [inline app/assets/javascripts/components/post.js.jsx]: remove this

File: app/assets/javascripts/components/score.js.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+var API = require('../api_stub');
+var Dispatcher = require('../flux/dispatcher');
+var ImageStore = require('../flux/image_store');
+var StateFromStore = require('react-components/state-from-store-mixin');
+var classNames = require('classnames');
+
+var Score = React.createClass({
+  getInitialState: function() {
+    return {
+      upClicked: false,
```
Comment (bradencanderson) [inline app/assets/javascripts/components/score.js.jsx:10]: Use StateFromStore to bring state into a component. The way you have it is if Score is unmounted, we lose state.

File: app/assets/javascripts/flux/image_store.js.jsx [hunk 3]
```diff
@@ -49,6 +58,53 @@ class ImageStore {
       this._listeners.splice(index, 1);
     }
   }
+
+  upvote(id) {
+    console.log(id);
```
Comment (bradencanderson) [inline app/assets/javascripts/flux/image_store.js.jsx]: remove this

File: app/assets/javascripts/flux/image_store.js.jsx [hunk 4]
```diff
@@ -49,6 +58,53 @@ class ImageStore {
...
+      id,
+      (success) => {
+        console.log("success");
+      },
+      (error) => {
+        console.error('Error upvote');
+      }
+    );
+      if (this.isDownvoted(id)) {
+        my_image.score++;
```
Comment (bradencanderson) [inline app/assets/javascripts/flux/image_store.js.jsx]: Shouldn't you set `my_image.downvoted` to False here? Or even better, use a ternary value (`status=oneOf("UP", "DOWN", "NONE")`) to show that these states are exclusive.

File: app/assets/javascripts/flux/image_store.js.jsx [hunk 5]
```diff
@@ -49,6 +58,53 @@ class ImageStore {
...
+    API.postDownvote(
+      id,
+      (success) => {
+        console.log("success");
+      },
+      (error) => {
+        console.error('Error downvote');
+      }
+    );
+      if (this.isUpvoted(id)) {
```
Comment (bradencanderson) [inline app/assets/javascripts/flux/image_store.js.jsx]: indents

File: app/assets/javascripts/flux/image_store.js.jsx [hunk 6]
```diff
@@ -49,6 +58,53 @@ class ImageStore {
...
+        my_image.score--;
+      }
+      my_image.score--;
+      this._notify();
+  }
+
+  isUpvoted(id) {
+    return _.find(this._images, function(obj) { return obj.id == id }).upvoted;
```
Comment (bradencanderson) [inline app/assets/javascripts/flux/image_store.js.jsx:118]: Try to handle the "id not found" case also.

File: db/schema.rb [hunk 7]
```diff
@@ -37,13 +37,6 @@
     t.datetime "img_updated_at"
   end
 
-  create_table "posts", force: :cascade do |t|
```
Comment (bradencanderson) [inline db/schema.rb:40]: What is this?

**Comment by bradencanderson** [comment]
This is good and probably ready to merge. I just want to make sure component state is persisted when we navigate between views and components get unmounted. Putting app state in a Store is a good way to solve that issue.

File: db/schema.rb (hunk 7 above)
Comment (pcrumm) [inline db/schema.rb:40]: @bradencanderson  removing the posts table, which we no longer use. Probably not worth asking questions about this file as it's automatically generated.

**Pull request merged**